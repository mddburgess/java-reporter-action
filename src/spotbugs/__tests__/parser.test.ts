import * as glob from "@actions/glob";
import SpotbugsParser from "../parser";

describe("SpotbugsParser", () => {
  it("can parse an empty SpotBugs report", () => {
    const report = new SpotbugsParser("src/spotbugs/__tests__/__fixtures__/empty.xml").read();
    expect(report).toMatchSnapshot();
  });

  it("can parse a real SpotBugs report", async () => {
    const globber = await glob.create("**/target/spotbugsXml.xml");
    const paths = await globber.glob();
    expect(paths.length).toBeGreaterThan(0);

    const report = new SpotbugsParser(paths[0]).read();
    expect(report).toMatchSnapshot();
  });
});
