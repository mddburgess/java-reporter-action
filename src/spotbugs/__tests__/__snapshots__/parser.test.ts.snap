// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotbugsParser can parse a real SpotBugs report 1`] = `
Object {
  "bugs": Array [
    Object {
      "category": "MALICIOUS_CODE",
      "endLine": 11,
      "filePath": "spotbugs/maliciouscode/EiExposeRep.java",
      "longMessage": "spotbugs.maliciouscode.EiExposeRep.getArrayVar() may expose internal representation by returning EiExposeRep.arrayVar",
      "priority": 2,
      "shortMessage": "May expose internal representation by returning reference to mutable object",
      "startLine": 11,
    },
    Object {
      "category": "MALICIOUS_CODE",
      "endLine": 15,
      "filePath": "spotbugs/maliciouscode/EiExposeRep.java",
      "longMessage": "spotbugs.maliciouscode.EiExposeRep.setArrayVar(String[]) may expose internal representation by storing an externally mutable object into EiExposeRep.arrayVar",
      "priority": 2,
      "shortMessage": "May expose internal representation by incorporating reference to mutable object",
      "startLine": 15,
    },
  ],
  "categories": Map {
    "MALICIOUS_CODE" => "Malicious code vulnerability",
  },
}
`;

exports[`SpotbugsParser can parse an empty SpotBugs report 1`] = `
Object {
  "bugs": Array [],
  "categories": Map {},
}
`;
